src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -195,24 +192,62 @@ public static class TarUtils
             }
         }
 
-        private static void SetProcessStartInfo(Process process, string processFileName, string processArguments)
+        private static void SetProcessStartInfo(ProcessStartInfo processStartInfo, string processFileName, string processArguments, string processWorkingDirectory)
         {
-            process.StartInfo.FileName = processFileName;
-            process.StartInfo.Arguments = processArguments;
-            process.StartInfo.UseShellExecute = false;
-            process.StartInfo.RedirectStandardInput = true;
-            process.StartInfo.RedirectStandardOutput = true;
-            process.StartInfo.RedirectStandardError = true;
+            processStartInfo.FileName = processFileName;
+            processStartInfo.Arguments = processArguments;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.RedirectStandardInput = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.WorkingDirectory = processWorkingDirectory;
+        }
+
+        private static ProcessStartInfo GetCreateTarProcessInfo(string archiveFileName, string inputPath)
+        {
+            var processFileName = "tar";
+            inputPath = (inputPath.EndsWith(Path.DirectorySeparatorChar)) ? inputPath.TrimEnd(Path.DirectorySeparatorChar) : inputPath;
+            var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path for the '-cf' option, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar' for that option.
+            ProcessStartInfo processStartInfo = new ProcessStartInfo();
+            SetProcessStartInfo(processStartInfo, processFileName, processArguments, processWorkingDirectory: Path.GetTempPath()); // We want to create the archiveFile in temp folder, and hence starting the tar process from TEMP to avoid absolute paths in tar cmd line.
+            return processStartInfo;
+        }
+
+        private static ProcessStartInfo GetExtractStartProcessInfo(string targetDirectory)
+        {
+            bool does7zExists = isWindows ? CheckIf7ZExists() : false;
+            string processFileName = (does7zExists) ? "7z" : "tar";
+            string processArguments = (does7zExists) ? $"x -si -aoa -o\"{targetDirectory}\" -ttar" : $"-xf - -C ."; // Instead of targetDirectory, we are providing . to tar, because the tar process is being started from targetDirectory.
```

Comment:
Do it this way to put 7z with 7z args and tar with tar args

 string processFileName;
 string processArguments;
if (isWindows && CheckIf7ZExists()) {
  processFileName = "7z";
  processArgumetns = $"x -si -aoa -o\"{targetDirectory}\" -ttar";
} else {

}